src/app/components/editor/editor.component.ts

```diff
@@ -82,6 +83,23 @@ export class EditorComponent {
         flexGrow: 1,
       },
     });
+
+    // Check for URL parameter and decode if present
+    this.route.queryParams.subscribe(params => {
+      const encodedContent = params['code'];
+      if (encodedContent) {
+        const decodedContent = atob(encodedContent);
```

Comment:
what happens if this throws can you test if it still responds to changes or does it unsubscribe

Reply:
![image](https://github.com/user-attachments/assets/07b219f8-1a78-4a8d-9456-1459ba14bff3)